File: backend/app/schemas/cliente.py
from pydantic import BaseModel, EmailStr
from typing import Optional
from datetime import datetime

class ClienteBase(BaseModel):
    nombre: str
    identificacion_fiscal: Optional[str] = None
    telefono: Optional[str] = None
    email: Optional[EmailStr] = None
    direccion: Optional[str] = None

class ClienteCreate(ClienteBase):
    pass

class ClienteUpdate(ClienteBase):
    pass

class ClienteOut(ClienteBase):
    cliente_id: int
    fecha_creacion: datetime
    fecha_actualizacion: Optional[datetime] = None

    class Config:
        from_attributes = True

File: types/cliente.ts
// types/cliente.ts

export interface ClienteBase {
  nombre: string;
  identificacion_fiscal?: string;
  telefono?: string;
  email?: string;
  direccion?: string;
}

export interface ClienteCreate extends ClienteBase {}

export interface ClienteUpdate extends ClienteBase {}

export interface ClienteOut extends ClienteBase {
  cliente_id: number;
  fecha_creacion: string; // Fecha en formato string (ISO 8601)
  fecha_actualizacion?: string;
}
